pkg/kvstore/configurable.go

```diff
@@ -75,23 +73,32 @@ func newKvStoreChainFromConfig(config cfg.Config, logger mon.Logger, name string
 			GetsPerPromote: configuration.InMemory.GetsPerPromote,
 		},
 	})
+	if err != nil {
+		return nil, fmt.Errorf("can not create chain store: %w", err)
+	}
 
 	for _, element := range configuration.Elements {
 		switch element {
 		case TypeDdb:
-			store.Add(NewDdbKvStore)
+			if err := store.Add(NewDdbKvStore); err != nil {
+				return nil, fmt.Errorf("can not add ddb store: %w", err)
+			}
 		case TypeInMemory:
-			store.Add(NewInMemoryKvStore)
+			if err := store.Add(NewInMemoryKvStore); err != nil {
+				return nil, fmt.Errorf("can not add inMemory store: %w", err)
+			}
 		case TypeRedis:
-			store.Add(NewRedisKvStore)
+			if err := store.Add(NewRedisKvStore); err != nil {
+				return nil, fmt.Errorf("can not add redis store: %w", err)
+			}
 		default:
 			err := fmt.Errorf("invalid element type %s for kvstore chain", element)
```

Comment:
Can't you pass the error back in the default cause instead of `Fatalf`ing?

Reply:
good catch